network/src/network.rs

```diff
@@ -76,13 +76,14 @@ pub struct SessionInfo {
 pub struct NetworkState {
     pub(crate) peer_registry: RwLock<PeerRegistry>,
     pub(crate) peer_store: Mutex<Box<dyn PeerStore>>,
-    pub(crate) original_listened_addresses: RwLock<Vec<Multiaddr>>,
-    dialing_addrs: RwLock<FnvHashMap<PeerId, Instant>>,
+    pub(crate) original_listened_addrs: RwLock<Vec<Multiaddr>>,
+    dialing_addrs: RwLock<HashMap<PeerId, Instant>>,
 
-    pub(crate) protocol_ids: RwLock<FnvHashSet<ProtocolId>>,
-    listened_addresses: RwLock<FnvHashMap<Multiaddr, u8>>,
+    pub(crate) protocol_ids: RwLock<HashSet<ProtocolId>>,
+    listened_addrs: RwLock<HashMap<Multiaddr, u8>>,
+    pending_advise_addrs: RwLock<HashSet<Multiaddr>>,
```

Comment:
What is an advise address? A weird term I have never seen before.

Reply:
I'll use term `observed address` - reference to external addresses that observed by remote peers.